sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1760,6 +1760,13 @@ object SQLConf {
     .internal()
     .intConf
     .createWithDefault(Int.MaxValue)
+
+  val LEGACY_CAST_DATE_TIMESTAMP_TO_STRING =
+    buildConf("spark.sql.legacy.binaryComparison.castDateTimestampToString")
+      .doc("If it is set to true, date/timestamp will cast to string in binary comparisons " +
```

Comment:
can we also add a migration guide for this behavior change and mention this config?